modules/util/portal-tools-rest-builder/src/main/resources/com/liferay/portal/tools/rest/builder/dependencies/base_resource_impl.ftl

```diff
@@ -165,6 +165,8 @@ public abstract class Base${schemaName}ResourceImpl
 					<#assign getBatchJavaMethodSignature = javaMethodSignature />
 				</#if>
 			</#if>
+		<#elseif stringUtil.equals(javaMethodSignature.methodName, "getSiteByExternalReferenceCode" + schemaNames + "Page")>
```

Comment:
I think you are just supporting the endpoints with parent site, but not parent asset library. I think both should be supported